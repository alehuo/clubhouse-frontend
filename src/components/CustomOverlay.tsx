import React from "react";
import { OverlayTrigger, Tooltip } from "react-bootstrap";

interface Props {
  id: string;
  delay?: number;
  text: string | React.ReactElement<any>;
  children: React.ReactElement<any>;
}

const CustomOverlay: React.SFC<Props> = ({ id, delay, text, children }) => {
  return (
    <OverlayTrigger
      placement="bottom"
      delay={delay ? delay : 150}
      overlay={<Tooltip id={id}>{text}</Tooltip>}
    >
      {children}
    </OverlayTrigger>
  );
};

export default CustomOverlay;
