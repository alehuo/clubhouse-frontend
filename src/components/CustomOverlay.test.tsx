import { shallow } from "enzyme";
import React from "react";
import CustomOverlay from "./CustomOverlay";

it("CustomOverlay matches snapshot", () => {
  const cmpnt = shallow(
    <CustomOverlay id="test" delay={250} key="HelloWorld" text="Testing">
      <>Overlay text</>
    </CustomOverlay>,
  );
  expect(cmpnt).toMatchSnapshot();
});
