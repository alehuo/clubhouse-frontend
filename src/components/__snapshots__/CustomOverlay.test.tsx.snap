// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomOverlay matches snapshot 1`] = `ShallowWrapper {}`;
